Katello/katello
# Fixes #31740 - katello_events and candlepin_events intermittently show FAIL/not started

This PR includes some chores and refactoring, and also attempts to fix ~~a bug~~ two bugs.

### chores

* move `event_daemon.rb` to `event_daemon/runner.rb`
* move `Katello::EventDaemon` to `Katello::EventDaemon::Runner`
* move `Katello::EventDaemon::Monitor` to its own file
* move all `Rails.cache` use to `Katello::EventDaemon::Runner`

### how we fix the actual problem

* Don't clear `Rails.cache` on `stop`.  thanks @jturel for inspiration

### how to test

1. In `app/models/katello/ping.rb`, add the following at the top of the `ping` method:
```rb
        if Katello::EventDaemon::Runner.pid && Process.pid != Katello::EventDaemon::Runner.pid
          Process.kill(3, Katello::EventDaemon::Runner.pid)
          sleep(10)
        end
```
2. From another tab, `curl localhost:3000/api/v2/ping` until it sleeps
3. If needed, ping while it's sleeping from a third tab
4. You should see it come back FAIL / not running
5. Now apply the patch and do it again
